tests/suite-unit/PluginFormcreatorForm.php

```diff
@@ -445,4 +445,70 @@ public function testgetFormFromSection($section, $expected) {
       $output = $form->getFromDBBySection($section);
       $this->boolean($output)->isEqualTo($expected);
    }
+
+   public function testSaveForm() {
+      global $CFG_GLPI;
+
+      $this->login('glpi', 'glpi');
+
+      // disable notifications as we may fail in some case (not the purpose of this test btw)
+      $use_notifications = $CFG_GLPI['use_notifications'];
+      $CFG_GLPI['use_notifications'] = 0;
+
+      $answer = "test answer to question";
+
+      // prepare a fake form with targets
+      $question = $this->getQuestion();
+      $section = new \PluginFormcreatorSection();
+      $section->getFromDB($question->fields[\PluginFormcreatorSection::getForeignKeyField()]);
+      $form = new \PluginFormcreatorForm();
+      $form->getFromDB($section->fields[\PluginFormcreatorForm::getForeignKeyField()]);
+      $formFk = \PluginFormcreatorForm::getForeignKeyField();
+      $this->getTargetTicket([
+         $formFk => $form->getID(),
+      ]);
+      $this->getTargetChange([
+         $formFk => $form->getID(),
+      ]);
+
+      // prepare input
+      $input = [
+         'formcreator_form' => $form->getID(),
+         'formcreator_field_'.$question->getID() => $answer
+      ];
+
+      // send for answer
+      $formAnswerId = $form->saveForm($input);
+      $this->boolean((bool)$formAnswerId)->isTrue();
```

Comment:
if a success is expected it is probably better ti check that $formAnswerId is an integer ?

(the method returns a bool only on failure)

Reply:
the cast bool do the same thing, but i can if you prefer

Reply:
this is how we do in GLPI test, we don't care to check the integer, the goal is indeed to test if we don't have a false return

Reply:
The following should to the job as well

```suggestion
      $this->integer($formAnswerId);
```